verifyFiles.py

```diff
@@ -181,7 +216,42 @@ def n_differences_across_subjects(conditions_dict,root_dir,metrics,checksums_fro
                                 if file_name not in metric_values[metric['name']][key].keys():
                                     metric_values[metric['name']][key][file_name]=0
                                 metric_values[metric['name']][key][file_name] += float(run_command(metric['command'],file_name,c,d,subject,root_dir))
-    return diff,metric_values
+                        # if we are in different runs of the same
+                        # condition (see previous comment) then
+                        # inspect the reprozip trace here to get the
+                        # list of executables that created such
+                        # differences
+		        if sqlite_db_path and files_are_different:
+			  #Monitor.txt seems not to have entry in sqlite table
+			  if file_name not in dictionary_executables:
+			   dictionary_executables[file_name]=get_executable_details(conn,sqlite_db_path,file_name,is_intra_condition_run)
+    if sqlite_db_path:
+      conn.close()
+    return diff,metric_values,dictionary_executables
+
+#Method get_executable_details is used for finding out the details of the processes that created or modified the specified file.
+def get_executable_details(conn,sqlite_db_path,file_name,is_intra_condition_run):#TODO Intra condition run is not taken into account while the executable details are getting written to the file
+    sqlite_cursor = conn.cursor()
+    #opened_files table has a column named MODE
+    # The definition of the mode values are as described below
+    #FILE_READ =1
+    #FILE_WRITE=2
+    #FILE_WDIR=4
+    #FILE_STAT=8
+    #FILE_LINK=16
+    sqlite_cursor.execute('SELECT DISTINCT executed_files.name,executed_files.argv,executed_files.envp,executed_files.timestamp,executed_files.workingdir from executed_files INNER JOIN opened_files where opened_files.process = executed_files.process and opened_files.name like ? and opened_files.mode=2 and opened_files.is_directory=0',('%/'+file_name,))
+    data = sqlite_cursor.fetchall()
+    sqlite_cursor.close()    
+    executable_details_list=[]
+    if data:
+      if is_intra_condition_run:
```

Comment:
This is a bit weird because the content of executable_details_list now depends on the order on which you will process the conditions. For instance, if a file has both intra-condition and inter-condition differences, the flag added to the dictionary will be 'true' if intra-condition differences are found first and 'false' otherwise. I would remove the flag from this method and just preprend `*` to `key` on line 152 so that intra-condition differences are easily identified.